sql-plugin/src/main/scala/com/nvidia/spark/rapids/spill/SpillFramework.scala

```diff
@@ -338,24 +405,32 @@ class SpillableHostBufferHandle private (
                 }
               }
             }
-            disk = Some(diskHandleBuilder.build)
-            sizeInBytes
+            var staging: Option[DiskHandle] = Some(diskHandleBuilder.build)
+            synchronized {
+              spilling = false
+              if (closed) {
+                staging.foreach(_.close())
+                staging = None
+                doClose()
+              } else {
+                disk = staging
+              }
+              releaseHostResource()
```

Comment:
this is calling into the stores again with the handle lock held. Does it need to be in this synchronized?

Reply:
good catch, i think not, will fix